sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -260,7 +287,28 @@ async def serialize_property(value: 'Input[Any]',
         # resolved with known values.
         is_known = await output._is_known
         is_secret = await output._is_secret
-        value = await serialize_property(output.future(), deps, input_transformer, typ)
+        value = await serialize_property(output.future(), deps, input_transformer, typ, keep_output_values=False)
+
+        if keep_output_values and await settings.monitor_supports_output_values():
+            # TODO expand the dependencies
+            dependencies: Set[str] = set()
+            for resource in value_resources:
+                urn = await serialize_property(resource.urn, deps, input_transformer, keep_output_values=False)
+                dependencies.add(cast(str, urn))
```

Comment:
Similar question here as https://github.com/pulumi/pulumi/pull/7925/files#discussion_r704783076

Reply:
Actually, I'm going to do this as a quick follow-up. The resource cycle breaker added in #7887 makes adopting this inside the RPC code a little more difficult because it needs a `from_resource: 'Resource'`. I opened https://github.com/pulumi/pulumi/issues/7977 and updated the TODO to link to the issue.